File: Step 3: Solve Problems on Arrays/Step 3.1: Easy/LeftRotate.cpp
#include <bits/stdc++.h>
using namespace std;

// Follow the steps below to solve the given problem.

//     Rotate the array to left by one position. For that do the following:
//         Store the first element of the array in a temporary variable.
//         Shift the rest of the elements in the original array by one place.
//         Update the last index of the array with the temporary variable.
//     Repeat the above steps for the number of left rotations required.

/*Function to left rotate arr[] of size n by d*/
void Rotate(int arr[], int d, int n) {
    int p = 0;
    while (p < d) {
        int last = arr[0];
        for (int i = 0; i < n - 1; i++) {
            arr[i] = arr[i + 1];
        }
        arr[n - 1] = last;
        p++;
    }
}

// Driver code
int main() {
    int arr[] = {1, 2, 3, 4, 5, 6, 7};
    int N = sizeof(arr) / sizeof(arr[0]);
    int d = 2;

    // Function calling
    Rotate(arr, d, N);
    for (int i = 0; i < N; i++) cout << arr[i] << " ";

    return 0;
}

File: Step 3: Solve Problems on Arrays/Step 3.1: Easy/LeftRotate.py
# Python program to rotate an array by d elements

# Follow the steps below to solve the given problem.

#     Rotate the array to left by one position. For that do the following:
#         Store the first element of the array in a temporary variable.
#         Shift the rest of the elements in the original array by one place.
#         Update the last index of the array with the temporary variable.
#     Repeat the above steps for the number of left rotations required.


# Function to left rotate arr[] of size n by d
def Rotate(arr, d, n):
    p = 0
    while (p < d):
        last = arr[0]
        for i in range(n - 1):
            arr[i] = arr[i + 1]
        arr[n - 1] = last
        p = p + 1


# Driver code
arr = [1, 2, 3, 4, 5, 6, 7]
N = len(arr)
d = 2

# Function calling
Rotate(arr, d, N)
for i in range(N):
    print(arr[i], end=" ")
